Zend/zend_object_handlers.c

```diff
@@ -819,7 +819,20 @@ ZEND_API zval *zend_std_write_property(zend_object *zobj, zend_string *name, zva
 				}
 
 				ZVAL_COPY_VALUE(&tmp, value);
-				if (UNEXPECTED(!zend_verify_property_type(prop_info, &tmp, property_uses_strict_types()))) {
+				// Increase refcount to prevent object from being released in __toString()
+				GC_ADDREF(zobj);
+				bool type_matched = zend_verify_property_type(prop_info, &tmp, property_uses_strict_types());
+				if (UNEXPECTED(GC_DELREF(zobj) == 0)) {
+					zend_throw_error(NULL, "Object was released while assigning property %s::$%s",
+						ZSTR_VAL(prop_info->ce->name), zend_get_unmangled_property_name(prop_info->name));
```

Comment:
I'm not sure if we should throw an exception or silently ignore the problem. Both decisions are acceptable.

Reply:
I would prefer silently ignoring it, but I'm not sure if that can work. `write_property` returns the assigned value which it does by returning a pointer to the property storage. That storage is freed by the end of `write_property` in this case though.

Reply:
Having an exception here feels weird, I would expect it to be silently ignored too. Wouldn't it be fine returning `value` directly in that case?

Reply:
I would prefer not change prototypes without a big reason.
Can't we use the value copied into ``tmp``?

Reply:
But in what storage? `tmp` is stack-allocated and can't be returned. `value` might be a CV, so it can't be used for coerced values. An option would be to allocate a new temporary value on call-site, copy `value` into it and then pass that to `write_property`. The downside is that we need a copy for all cases in the `ZEND_ASSIGN_OBJ`. If that's acceptable that should work.

Reply:
Then it's easy to throw an exception.

Reply:
Since there's no simple approach for returning the value I'm sticking with this approach. If somebody can come up with something more consistent, please let me know!